google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/AddCloudLibrariesDialog.java

```diff
@@ -126,6 +133,29 @@ protected void doOKAction() {
     }
   }
 
+  /**
+   * Fetches all the available roles for the {@link CloudProject} and returns the set of roles
+   * corresponding to the selected client libraries.
+   *
+   * @param apis the set of {@link CloudLibrary apis} selected
+   * @return the set of {@link Role roles} corresponding to the selected apis
+   */
+  private Set<Role> getServiceAccountRolesRoles(Set<CloudLibrary> apis) {
+    List<Role> serviceAccountRoles = CloudApiManager.getServiceAccountRoles(getCloudProject());
+
+    Set<String> roleIds =
+        apis.stream()
+            .map(CloudLibrary::getServiceRoles)
+            .filter(Objects::nonNull)
+            .flatMap(Collection::stream)
+            .collect(Collectors.toSet());
+
+    return serviceAccountRoles
+        .stream()
+        .filter(role -> roleIds.contains(role.getName()))
```

Comment:
Do we need `contains()` or `equals()` to match role names in this filter, can they be slightly different in names so `contains()` is required?

Reply:
I think it needs to be contains because roleIds is a list of roles. Did I misunderstand your point?

Reply:
Oops sorry, I forgot it's a list, not a string, sorry.